waspc/packages/deploy/src/providers/railway/README.md

```diff
@@ -0,0 +1,39 @@
+### Take into account things
+
+1. `railway login` does not work in non-interactive mode. Current implementation checks if user logged in, otherwise throw out of the process with a friendly hint what to do next
+2. Global options `--wasp-exe <path>` and `--wasp-project-dir <dir>` are being added to each subcommand in the global scope
+3. This implementation expects a specific file for railway deployments to be present in the main root directory of Wasp - `wasp.deployrc.railway.json`
+   It's expected signature is:
+
+```
+	"environment": "production",
+	"projectId": "bef41c2c-b1a9-42ef-adb9-e3afa84fbbea",
+	"clientService": {
+		"name": "client",
+		"url": "client-production-6ea1.up.railway.app"
+	},
+	"serverService": {
+		"name": "server",
+		"url": "server-production-dfaf.up.railway.app"
+	}
+```
+
+Environment - required by Railway as there can exist multiple envs
+ProjectID - build folder has to be linked with a respective project
+Client / Server Service - info about the client service (ID and its URL). Railways projects consist of services, that you can spin up pretty easily. Each service has its own env variables etc.
+
+Whenever "setup" feature is implemented, we can ensure that this file is create automatically, but I am not entirely sure if that's a good idea to do. Need peoplez thoughts on this. Alternative, we force users to provide those values as flags / arguments.. ?
+
+### Refactor suggestions things
+
+4. Suggesting to keep all types in `.d.ts` files to clearly indicate what they are
+5. Even though haven't done it properly myself, but I'd probably also remove Capital casing on type files, as I see no benefit in this, if we mark the file with `.d.ts` extension, it should be clear enough what file is for.
+6. Stick to naming convention of types to indicate where they belong to - FlyDeployOptions, Railway(Rw)DeployOptions. Eases search of specific options across the codebase instead of doing a global search and having 10+ same names
+7. Installed `fs-extra` package, hoping it's allowed :) otherwise, can be reimplemented (mainly used for `ensureDir`, but it provides other convenient stuffs)
```

Comment:
@shayneczyzewski Will this have a too negative effect our build size? It can stay, as far as I'm concerned.

Reply:
I think we will be ok there, seems pretty small 👍🏻